src/core/cluster.py

```diff
@@ -4,19 +4,20 @@
 
 """Collection of global cluster state for the ZooKeeper quorum."""
 import logging
-from typing import Set
+from typing import Dict, Set
 
+from charms.data_platform_libs.v0.data_interfaces import DataPeerOtherUnit
 from ops.framework import Framework, Object
-from ops.model import Relation
+from ops.model import Relation, Unit
 
-from core.models import SUBSTRATES, ZKClient, ZKCluster, ZKServer
+from core.models import SUBSTRATES, CharmWithRelationData, ZKClient, ZKCluster, ZKServer
 from literals import CLIENT_PORT, PEER, REL_NAME, SECURE_CLIENT_PORT
 
 logger = logging.getLogger(__name__)
 
 
-class ClusterState(Object):
-    """Collection of global cluster state for the ZooKeeper quorum."""
+class ClusterStateBase(Object):
+    """Collection of global cluster state for Framework/Object."""
```

Comment:
**nitpick:** If this is just a Base, I think we can put it in `src/core/models.py`